private/buf/bufgen/features.go

```diff
@@ -132,11 +134,22 @@ func checkRequiredFeatures(
 			sort.Slice(failedFeatures, func(i, j int) bool {
 				return failedFeatures[i] < failedFeatures[j]
 			})
+			// We only log failed features (whereas we error on failed editions). This is in
+			// keeping with CLI versions pre-1.32.0 (BSR-3931).
+			_, _ = fmt.Fprintf(stderrContainer.Stderr(), "Warning: plugin %q does not support required features.\n", pluginName)
```

Comment:
Oh, it was mostly because I was following the older pattern with the multi-line warning.
But I can adjust this to use `logger.Warn` and construct a single log-line.